Doc/library/os.path.rst

```diff
@@ -306,11 +306,13 @@ the :mod:`glob` module.)
 
    Join one or more path components intelligently.  The return value is the
    concatenation of *path* and any members of *\*paths* with exactly one
-   directory separator (``os.sep``) following each non-empty part except the
-   last, meaning that the result will only end in a separator if the last
-   part is empty.  If a component is an absolute path, all previous
-   components are thrown away and joining continues from the absolute path
-   component.
+   directory separator following each non-empty part except the last, meaning
+   that the result will only end in a separator if the last part is empty.  If
+   a component is an absolute path, all previous components are thrown away
+   and joining continues from the absolute path component.
+   
+   On Windows, the directory separator is a backslash, while on POSIX platforms
+   it is a forward slash.
```

Comment:
Leave this part out.

What the separator is affects many functions in `os.path`, not only `join()`.  If it needs to be documented for `os.path`, it's better to document `os.path.sep`, and to describe there that it's informative, and not intended to be modified.

Since this isn't a factor in bpo-43620, it should not be a part of this PR.